app/forms/manage_users_form.rb

```diff
@@ -14,28 +14,34 @@ def initialize(team)
   def submit(user_emails:)
     # Devise makes emails case insensitive by downcasing prior to saving
     @user_emails = user_emails.map(&:downcase)
+    @emails_to_remove = team.users.pluck(:email) - @user_emails
+    @emails_to_add = @user_emails - team.users.pluck(:email)
 
     return false unless valid?
 
-    result = team.update(users: users_from_user_emails)
-    errors.merge!(team.errors)
-    result
+    ActiveRecord::Base.transaction do
+      destroy_removed_user_teams
+      add_new_users
+    end
+    true
+  rescue ActiveRecord::ActiveRecordError => e
+    errors.add(:base, "Unable to update users: #{e.message}") 
```

Comment:
This may not play nicely with all `ActiveRecordError` children but it also should be _very_ unlikely. I can remove the inclusion of `e.message` to keep it simpler.

Reply:
No longer necessary